DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs

```diff
@@ -6,44 +6,73 @@
-        private IDeviceRegistryListRepository _deviceRegistryListRepositoryMock;
-        private IVirtualDeviceStorage _virtualDeviceStorageMock;
-        private IConfigurationProvider _configProviderMock;
-        private ISecurityKeyGenerator _securityKeyGeneratorMock;
-        private IDeviceRulesLogic _deviceRulesLogicMock;
-        private IDeviceLogic _deviceLogicMock;
+        private Mock<IIotHubRepository> _iotHubRepositoryMock;
+        private Mock<IDeviceRegistryCrudRepository> _deviceRegistryCrudRepositoryMock;
+        private Mock<IDeviceRegistryListRepository> _deviceRegistryListRepositoryMock;
+        private Mock<IVirtualDeviceStorage> _virtualDeviceStorageMock;
+        private Mock<IConfigurationProvider> _configProviderMock;
+        private Mock<ISecurityKeyGenerator> _securityKeyGeneratorMock;
+        private Mock<IDeviceRulesLogic> _deviceRulesLogicMock;
+        private IDeviceLogic _deviceLogic;
+        private Fixture fixture;
 
         public DeviceLogicTests()
         {
-            _iotHubRepositoryMock = (new Mock<IIotHubRepository>()).Object;
-            _deviceRegistryCrudRepositoryMock = (new Mock<IDeviceRegistryCrudRepository>()).Object;
-            _deviceRegistryListRepositoryMock = (new Mock<IDeviceRegistryListRepository>()).Object;
-            _virtualDeviceStorageMock = (new Mock<IVirtualDeviceStorage>()).Object;
-            _configProviderMock = (new Mock<IConfigurationProvider>()).Object;
-            _securityKeyGeneratorMock = (new Mock<ISecurityKeyGenerator>()).Object;
-            _deviceRulesLogicMock = (new Mock<IDeviceRulesLogic>()).Object;
+            _iotHubRepositoryMock = new Mock<IIotHubRepository>();
+            _deviceRegistryCrudRepositoryMock = new Mock<IDeviceRegistryCrudRepository>();
+            _deviceRegistryListRepositoryMock = new Mock<IDeviceRegistryListRepository>();
+            _virtualDeviceStorageMock = new Mock<IVirtualDeviceStorage>();
+            _configProviderMock = new Mock<IConfigurationProvider>();
+            _securityKeyGeneratorMock = new Mock<ISecurityKeyGenerator>();
+            _deviceRulesLogicMock = new Mock<IDeviceRulesLogic>();
+            _deviceLogic = new DeviceLogic(_iotHubRepositoryMock.Object, _deviceRegistryCrudRepositoryMock.Object, _deviceRegistryListRepositoryMock.Object, _virtualDeviceStorageMock.Object, _securityKeyGeneratorMock.Object, _configProviderMock.Object, _deviceRulesLogicMock.Object);
+            fixture = new Fixture();
         }
 
-        ~DeviceLogicTests() { }
+        [Fact]
+        public async void GetDevicesTest()
+        {
+            DeviceListQuery q = fixture.Create<DeviceListQuery>();
+            DeviceListQueryResult r = fixture.Create<DeviceListQueryResult>();
+            _deviceRegistryListRepositoryMock.SetupSequence(x => x.GetDeviceList(It.IsAny<DeviceListQuery>()))
+                .ReturnsAsync(r)
+                .ReturnsAsync(new DeviceListQueryResult());
```

Comment:
I am using SetupSequence, so I can send different return values for recurring function calls
I can actually wanted to do a condition based returns, but lambda functions kept giving an error, so ended up using this.